lib/Backend/BailOut.cpp

```diff
@@ -1437,7 +1437,7 @@ BailOutRecord::BailOutHelper(Js::JavascriptCallStackLayout * layout, Js::ScriptF
     Js::InterpreterStackFrame* newInstance = nullptr;
     Js::Var* allocation = nullptr;
 
-    if (executeFunction->IsGenerator())
+    if (executeFunction->IsGenerator() || executeFunction->IsAsync())
```

Comment:
IsGenerator || IsAsync comes up a lot.

I recommend we combine these into a new IsCoroutine() method